packages/service-library/src/servicelib/long_running_tasks/_task.py

```diff
@@ -292,17 +304,17 @@ def start_task(
     task_name = f"{handler_module_name}.{handler.__name__}"
 
     # only one unique task can be running
-    if unique and task_manager.is_task_name_running(task_name):
-        managed_tasks_ids = list(task_manager.tasks[task_name].keys())
+    if unique and tasks_manager.is_task_running(task_name):
+        managed_tasks_ids = list(tasks_manager.get_task_group(task_name).keys())
         assert len(managed_tasks_ids) == 1  # nosec
-        managed_task = task_manager.tasks[task_name][managed_tasks_ids[0]]
+        managed_task = tasks_manager.get_task_group(task_name)[managed_tasks_ids[0]]
         raise TaskAlreadyRunningError(task_name=task_name, managed_task=managed_task)
 
     task_progress = TaskProgress.create()
     awaitable = handler(task_progress, **kwargs)
```

Comment:
proposal here (I actually have it in a PR, but feel free to modify or not take it if you don't like it):
```python
async def _progress_task(progress: TaskProgress, handler: Callable[..., Awaitable]):
        task_progress.publish(message="starting", percent=0)
        try:
            return await handler(task_progress, **kwargs)
        finally:
            # TODO: change that signature. it actually does not publish anything
            # and it can raise if percent is <0 or >1!! -> simplify
            task_progress.publish(message="finished", percent=1)

    task = asyncio.create_task(
        _progress_task(task_progress, handler), name=f"{__name__}.{task_name}"
    )
```

Reply:
so you do not need to put the percent 0 and 1 everywhere...

Reply:
that is a bigger change ... i did some preamble to simplify ``TaskProgress`` . I suggest that you adapt this change and add it in your PR

Reply:
The idea here was for whomever is implementing the task to update it's progress. Imposing a starting and finished percent could work. ti would make it simpler to tell if a task finished.

Reply:
ok